### ISSUE TITLE
Wrong number of components ?

### ISSUE THREAD

**Issue by craff** [ofloveandhate/bertini_real#76]
The following file should produce a sextic plane curve with 11 real components. Even
speciying a large sphere, bertini_real only gives 1 component. Unfortunatly I do not have matlab, so I can not see
the curve.

```
CONFIG

tracktype: 1;
mptype: 2;
END;

INPUT

variable_group x,y;
function f1;
f1= 5902958103587057/295147905179352825856*x^6 + 40531964440273725/72057594037927936*x^4*y^2 + 25036959739963005/72057594037927936*x^4*y + 1920267795234525/36028797018963968*
x^4 + -5/288230376151711744*x^3*y^2 + 1/36028797018963968*x^3*y + 1/288230376151711744*x^3 + -27015689041511865/72057594037927936*x^2*y^4 + 4173100506748115/18014398509481984
*x^2*y^3 + 7681560821320809/72057594037927936*x^2*y^2 + -16171977850148559/576460752303423488*x^2*y + -4618832124818515/576460752303423488*x^2 + -5/288230376151711744*x*y^4 +
 -1/1152921504606846976*x*y^3 + 3/576460752303423488*x*y^2 + -19/36893488147419103232*x + 4505040779251255/72057594037927936*y^6 + -4173335654696659/36028797018963968*y^5 + 1
920512623176073/36028797018963968*y^4 + 673871944243585/72057594037927936*y^3 + -2309574197692173/288230376151711744*y^2 + 3172797267229483/73786976294838206464;

END;
```

**Comment by craff** [comment]
I could use python ... That is not well advertised and took time to find. Everything is fine.

**Comment by ofloveandhate** [comment]
> I could use python ... That is not well advertised and took time to find

Indeed, I could do better at advertising the Python visualization suite.  There's documentation on it at https://doc.bertinireal.com/python/ .  It's not near as feature-rich as the Matlab suite, but it's a decent start.

**Comment by ofloveandhate** [comment]
It sounds like Bertini_real computed your curve correctly.  Is that right?

**Comment by craff** [comment]
Yes, I was just misleaded by a message saying one component. And the lack of visibility of the python interface. 

I would suggest to advertise both python and MATLAB from the web page. 

Thanks for your answer. 

Le 30 octobre 2022 12:33:42 GMT-10:00, silviana amethyst ***@***.***> a écrit :

**Comment by craff** [comment]
Hello,

By the way, the following file produces some points which are not attached to
any components and strange error messages:

found that edge XX has NEW leftpoint, but \nexists edge w point YY as right point.

Should I report an issue for this ?

Moreover I have many examples like this one, which are smooth but maximal (for
the number of components of that degree) or near to maximal. Are you
interested by a pull request with those examples ?

Cheers,
Christophe

--
Christophe Raffalli
tél: +689 87 23 11 48
web: http://raffalli.eu

CONFIG

tracktype: 1;
mptype: 2;
END;

INPUT

variable_group x,y;
function f1;
f1=4802220442246831/19807040628566084398385987584*x^9 + -32957377069502151/36893488147419103232*x^8*y + -4127278184210831/4611686018427387904*x^8 + 18782903670303897/38685626227668133590597632*x^7*y^2 + 7561630540176637/9671406556917033397649408*x^7*y + 5693178108211387/19342813113834066795298816*x^7 + -1705663850911881/1125899906842624*x^6*y^3 + -18902405837490917/4503599627370496*x^6*y^2 + -8651810759552779/2251799813685248*x^6*y + -1305953794057393/1125899906842624*x^6 + 315011533900007007/2475880078570760549798248448*x^5*y^4 + -40345820963913311/154742504910672534362390528*x^5*y^3 + -7733228406037869/2417851639229258349412352*x^5*y^2 + -14816189260530513/4835703278458516698824704*x^5*y + -4761853809220271/19342813113834066795298816*x^5 + 217724187343243077/144115188075855872*x^4*y^5 + -202227556944321475/144115188075855872*x^4*y^4 + -230714953530856375/36028797018963968*x^4*y^3 + 42224417403955545/72057594037927936*x^4*y^2 + 16188409727303585/2251799813685248*x^4*y + 7010874785145309/2251799813685248*x^4 + -149847998295597021/618970019642690137449562112*x^3*y^6 + 50909092104764579/38685626227668133590597632*x^3*y^5 + -66032171627363725/154742504910672534362390528*x^3*y^4 + -21671473069935465/4835703278458516698824704*x^3*y^3 + 1800901263000505/604462909807314587353088*x^3*y^2 + 6072845010747979/2417851639229258349412352*x^3*y + -3602636171081631/1208925819614629174706176*x^3 + -72660555599077215/144115188075855872*x^2*y^7 + 20952515247969515/9007199254740992*x^2*y^6 + -92285981401025601/72057594037927936*x^2*y^5 + -55756853188066755/9007199254740992*x^2*y^4 + 21584546303735815/4503599627370496*x^2*y^3 + 1752718696013841/281474976710656*x^2*y^2 + -24661532116679577/9007199254740992*x^2*y + -2356447604712079/1125899906842624*x^2 + 3797823682430613/77371252455336267181195264*x*y^8 + -2103174241724667/4835703278458516698824704*x*y^7 + 48839586779937051/38685626227668133590597632*x*y^6 + -17517881742383247/19342813113834066795298816*x*y^5 + -19949958704814265/9671406556917033397649408*x*y^4 + 2655108108292009/604462909807314587353088*x*y^3 + -1448704890269541/604462909807314587353088*x*y^2 + -6026761286318673/2417851639229258349412352*x*y + 9000483531589545/2417851639229258349412352*x + 2029077078971371/36028797018963968*y^9 + -4205013010969583/9007199254740992*y^8 + 5767873835967367/4503599627370496*y^7 + -6378636657685255/9007199254740992*y^6 + -5396136573937911/2251799813685248*y^5 + 7010874785995565/2251799813685248*y^4 + 513781918292745/562949953421312*y^3 + -2356447606089655/1125899906842624*y^2 + 1289719790206729/1208925819614629174706176*y + 1123822840424051/144115188075855872;

END;

**Comment by ofloveandhate** [comment]
I would happily accept a PR with some more curve examples.  

I bet the issues you're having with "NEW" points will be solved by adjusting tolerances.  Bertini_real is numerical software, and thus "makes mistakes".  The following are not rare, especially without adjusting settings away from defaults:

* missing critical points, due to path crossings.  this can be improved by tightening `tracktolbeforeeg`, `tracktolduringeg`, and some others.
* mid-identifying points as the same or different from each other.  we're approximating points with floating point numbers, with finite accuracy, and so determining whether two points are the same is difficult.

i bet that adjusting a few settings makes bertini_real compute your curves correctly.  having them as part of my suite of test examples would be helpful.

~silviana

**Comment by ofloveandhate** [comment]
> I would suggest to advertise both python and MATLAB from the web page.

good suggestion.

**Comment by ofloveandhate** [comment]
> Yes, I was just misleaded by a message saying one component

Gotcha.  The number of components is reported by Bertini -- and is the number of *complex* components, not real.  It's common for a real curve (or surface) to have multiple "components", but in fact be parts of one complex object; thus, the real "components" aren't in fact separate components at all, just the real pieces of a single complex object.  

In the world of bertini_real, "component" is reserved for the complex algebraic variety language part of the theory, and we should use another word to describe the disconnected or singularly connected pieces of a real curve or surface.  I've been using the word "piece", particularly in the case of singularly connected pieces of a real algebraic surface.  If you have a better word than "piece", I'm open to it.

An elliptic curve with two pieces is an easy-to-see example.  Consider the curve $y^2=x^3-x$.  It has *one* complex component, with two disconnected real pieces: one is compact, and the other diverges to $\infty$.
